producer.go

```diff
@@ -255,10 +256,15 @@ func (opts *ProduceOpts) produce(p *Producer) error {
 	opts.MsgHeaders.MsgHeaders["$memphis_connectionId"] = []string{p.conn.ConnId}
 	opts.MsgHeaders.MsgHeaders["$memphis_producedBy"] = []string{p.Name}
 
+	data, err := p.validateMsg(opts.Message)
```

Comment:
check wether you need to validate, otherwise a lot of functions will be called

Reply:
there is a a check quite early inside that funciton
```
func (p *Producer) validateMsg(msg any) ([]byte, error) {
	sd, err := p.getSchemaDetails()
	if err != nil {
		return nil, err
	}

	if sd.schemaType == "" {
		switch msg.(type) {
		case []byte:
			return msg.([]byte), nil
		default:
			return nil, errors.New("Unsupported message type")
		}

	}

	return sd.validateProtoMsg(msg)
}

```